pkg/k8s/types/third_party.go

```diff
@@ -70,37 +70,45 @@ func (r *CiliumNetworkPolicy) Parse() (api.Rules, error) {
 		retRule.EndpointSelector.LabelSelector.MatchLabels[k8s.LabelSourceKeyPrefix+k8s.PodNamespaceLabel] = namespace
 	}
 
-	retRule.Ingress = make([]api.IngressRule, len(r.Spec.Ingress))
-	for i, ing := range r.Spec.Ingress {
-		retRule.Ingress[i].FromEndpoints = make([]api.EndpointSelector, len(ing.FromEndpoints))
-		for j, ep := range ing.FromEndpoints {
-			retRule.Ingress[i].FromEndpoints[j] = api.NewESFromK8sLabelSelector("", ep.LabelSelector)
-			if retRule.Ingress[i].FromEndpoints[j].MatchLabels == nil {
-				retRule.Ingress[i].FromEndpoints[j].MatchLabels = map[string]string{}
+	if r.Spec.Egress != nil {
```

Comment:
Is this correct? You are checking `r.Spec.Egress` for `!= nil` and then access `r.Spec.Ingress`